etcdmain/etcd.go

```diff
@@ -260,9 +260,16 @@ func startProxy(cfg *config) error {
 	}
 
 	cfg.ec.Dir = filepath.Join(cfg.ec.Dir, "proxy")
-	err = os.MkdirAll(cfg.ec.Dir, fileutil.PrivateDirMode)
-	if err != nil {
-		return err
+	if fileutil.Exist(cfg.ec.Dir) {
+		err := fileutil.CheckDirPermission(cfg.ec.Dir, fileutil.PrivateDirMode)
+		if err != nil {
+			return err
```

Comment:
> fmt.Errorf("%w: expect permissions %v", err, fileutil.PrivateDirMode)

will be much friendlier for end users

Reply:
It's being addressed in here https://github.com/etcd-io/etcd/pull/11997 Please feel free to add review comments there. Thanks!